### PR TITLE
sof: minor cleanups across source code

### PR DESCRIPTION
**Pull request by tlauda**

### CODE REVIEW

File: src/arch/xtensa/xtos/CMakeLists.txt [hunk 1]
```diff
@@ -11,10 +11,7 @@ set(VECTOR_DEFS
 	-D__SPLIT__level5
 )
 
-if(CONFIG_APOLLOLAKE
-   OR CONFIG_CANNONLAKE
-   OR CONFIG_SUECREEK
-   OR CONFIG_ICELAKE)
+if(CONFIG_LEVEL6_IRQ)
```
Comment (jajanusz) [inline src/arch/xtensa/xtos/CMakeLists.txt]: I think that ~20lines later you may also need to append 6 to `set(levels 2 3 4 5)`

File: src/include/ipc/control.h [hunk 2]
```diff
@@ -102,7 +101,7 @@ enum sof_ipc_ctrl_cmd {
 struct sof_ipc_ctrl_value_chan {
 	uint32_t channel;	/**< channel map - enum sof_ipc_chmap */
 	uint32_t value;
-} __packed;
+} __attribute__((packed));
```
Comment (jajanusz) [inline src/include/ipc/control.h]: Maybe we don't have to revert this. I'm wondering if it'd be good idea to make use of `-include` flag with header that has `__packed` etc so we can make these files to work like they were in kernel.
@mmaka1 @tlauda @lgirdwood @dbaluta What do you think?

File: src/include/sof/debug/debug.h [hunk 3]
```diff
@@ -28,7 +27,19 @@
...
+#define DEBUG_LOCKS	0
+#endif
+
+#if CONFIG_DEBUG_LOCKS_VERBOSE
+#define DEBUG_LOCKS_VERBOSE	1
+#else
+#define DEBUG_LOCKS_VERBOSE	0
+#endif
+
+#if CONFIG_DEBUG
```
Comment (jajanusz) [inline src/include/sof/debug/debug.h]: Why do we need aliases for those? Maybe it's better to keep them all with CONFIG_ prefixes and remove alias that was there for GDB.

File: src/include/sof/lib/cache.h [hunk 4]
```diff
@@ -22,4 +22,11 @@
 /* invalidate data */
 #define CACHE_INVALIDATE	1
 
+/* data cache line alignment */
+#if DCACHE_LINE_SIZE > 0
+#define PLATFORM_DCACHE_ALIGN	DCACHE_LINE_SIZE
+#else
+#define PLATFORM_DCACHE_ALIGN	4
```
Comment (jajanusz) [inline src/include/sof/lib/cache.h]: Why such a value?

Comment (tlauda) [reply src/include/sof/lib/cache.h]: sizeof(uint32_t), but I would like to have the possibility to include cache.h from assembly.

File: src/include/sof/debug/debug.h (hunk 3 above)
Comment (tlauda) [reply src/include/sof/debug/debug.h]: Then I cannot simply use it in a macro, because it doesn't always exist.

File: src/platform/imx8/lib/memory.c [hunk 5]
```diff
@@ -10,9 +10,6 @@
 #include <sof/platform.h>
 #include <ipc/topology.h>
 
-STATIC_ASSERT(0 == (HEAP_BUF_ALIGNMENT % PLATFORM_DCACHE_ALIGN),
```
Comment (jajanusz) [inline src/platform/imx8/lib/memory.c]: Why did you remove this?

File: src/arch/xtensa/xtos/CMakeLists.txt (hunk 1 above)
Comment (tlauda) [reply src/arch/xtensa/xtos/CMakeLists.txt]: I assumed it is used, but now I see we don't have level 6. I will double check and possibly remove it at all.

File: src/platform/imx8/lib/memory.c (hunk 5 above)
Comment (tlauda) [reply src/platform/imx8/lib/memory.c]: It doesn't make sense anymore and to be honest it didn't make sense before.

File: src/arch/xtensa/xtos/CMakeLists.txt (hunk 1 above)
Comment (jajanusz) [reply src/arch/xtensa/xtos/CMakeLists.txt]: Yea, these defines for lvl6 were there from autoconf times, but it looked that no1 was using it anyway. APL+ / CNL+ had defines but only Suecreek built lvl6 lib actually, but never included it into final binary tho.

File: src/include/sof/debug/debug.h (hunk 3 above)
Comment (jajanusz) [reply src/include/sof/debug/debug.h]: In which macro f.e.?

Comment (tlauda) [reply src/include/sof/debug/debug.h]: DEBUG_SET_FW_READY_FLAGS

Comment (jajanusz) [reply src/include/sof/debug/debug.h]: I would prefer to not have 2 symbols representing the same thing. You can play with #ifs or create macro that converts CONFIGs to bools that can be used f.e. in ternary operators.

File: src/include/ipc/control.h (hunk 2 above)
Comment (tlauda) [reply src/include/ipc/control.h]: The include would be hidden inside the make files, so it would be problematic if some external user app would like to take header "as is". With stdint we can at least assume that everyone has such header.

File: src/include/sof/lib/cache.h (hunk 4 above)
Comment (dbaluta) [reply src/include/sof/lib/cache.h]: @tlauda please provide a proper description of why this change is needed. I understand that for someone that is familiar with the code this is straightforward. Anyhow, it help help reviewers to rapidly understand your code. Also, please mention in the commit message that you prefer to define PLATFORM_DCACHE_ALIGN to 4 because of including it in assembly files.

File: src/arch/xtensa/include/arch/lib/cache.h [hunk 6]
```diff
@@ -75,6 +78,8 @@ static inline void dcache_writeback_invalidate_all(void)
 
 #endif /* __ARCH_LIB_CACHE_H__ */
```
Comment (dbaluta) [inline src/arch/xtensa/include/arch/lib/cache.h]: should this comment __ARCH_LIB_CACHE_H be moved to the endif below? Also maybe it worth to also add a comment about !_ASSEMBLER && !LINKER after endif.

File: src/platform/imx8/lib/memory.c (hunk 5 above)
Comment (dbaluta) [reply src/platform/imx8/lib/memory.c]: @tlauda either mention this in the commit message or make a separate commit. This is totally unrelated with the subject of the commit.

Also, please always explain why a commit is needed in the commit message!

File: src/arch/xtensa/xtos/CMakeLists.txt [hunk 7]
```diff
@@ -11,15 +11,7 @@ set(VECTOR_DEFS
...
 
-if(CONFIG_APOLLOLAKE
-   OR CONFIG_CANNONLAKE
-   OR CONFIG_SUECREEK
-   OR CONFIG_ICELAKE)
-	set(VECTOR_DEFS ${VECTOR_DEFS} -D__SPLIT__level6)
-endif()
-
 # Builds lib for each level from the same source files
-# TODO: should there be level 6 for apl+ ?
```
Comment (dbaluta) [inline src/arch/xtensa/xtos/CMakeLists.txt]: I really like that this commit message explains why the change is needed!

File: src/include/sof/debug/debug.h [hunk 8]
```diff
@@ -28,6 +27,18 @@
...
+#else
+#define DEBUG_LOCKS	0
+#endif
+
+#if CONFIG_DEBUG_LOCKS_VERBOSE
+#define DEBUG_LOCKS_VERBOSE	1
+#else
+#define DEBUG_LOCKS_VERBOSE	0
+#endif
+
```
Comment (dbaluta) [inline src/include/sof/debug/debug.h]: Why do you still need DEBUG_LOCKS and DEBUG_LOCKS_VERBOSE if you have the CONFIG_ version?

File: src/include/sof/lib/cache.h (hunk 4 above)
Comment (jajanusz) [reply src/include/sof/lib/cache.h]: Wouldn't something like sizeof(void*) be better than specific value?

File: src/include/sof/debug/debug.h (hunk 3 above)
Comment (jajanusz) [reply src/include/sof/debug/debug.h]: @tlauda @dbaluta  I mentioned in the past that we need something like IS_ENABLED etc stuff related to kconfig (look here: https://github.com/torvalds/linux/blob/master/include/linux/kconfig.h) with something like that you would be able to just use `IS_ENABLED(DEBUG)?a:b`. However I wasn't sure how to handle this because idk how we are going to handle modules...

**Comment by tlauda** [comment]
@dbaluta @jajanusz Updated.

**Comment by marc-hb** [comment]
> config: replace #ifdef with #if for CONFIG_* defines

While this is an important fix for non-booleans (if any), it doesn't make sense to me for Kconfig booleans that Kconfig never defines to zero https://github.com/ulfalizer/Kconfiglib#using-config-files-as-make-input
I don't think I ever saw Kconfig booleans used this way, @tlauda can you elaborate? The commit message of 3860b90c9e is empty. 

I tried to use `-Wundef` to trace some kconfig #include bug (see #3021 commit message) but the -Wundef noise from this commit makes `-Wundef | grep CONFIG_` unusable.

**Comment by jajanusz** [comment]
@marc-hb You are right that kconfig doesn't define them at all, but it's common practice to use `#if` instead of `#ifdef` to not have things like `#ifdef DEBUG` and you have `#define DEBUG 0` what makes no sense ;)

I think that in kernel both forms are used and we just used `#if` for consistency. 
Also I thought it had no disadvantages unless you showed issue with `-Wundef`.
However it's not that useful to change now all `#if CONFIG` to `#ifdef CONFIG`, because it will not f.e. find places in which we use `CONFIG` symbols without `config.h` header (cos `-Wundef` will not warn about symbols used with `#ifdef`) - it will help with just a few non-bool configs we have.
